packages/website/ts/components/legacy_portal/legacy_portal.tsx

```diff
@@ -86,7 +87,7 @@ export class LegacyPortal extends React.Component<LegacyPortalProps, LegacyPorta
     }
     constructor(props: LegacyPortalProps) {
         super(props);
-        this._sharedOrderIfExists = this._getSharedOrderIfExists();
+        this._sharedOrderIfExists = orderParser.parse(window.location.search);
```

Comment:
I don't think this will lead to bugs, but it may be better to use `withRouter` ? 
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/withRouter.md

The only way this could lead to a bug is if somehow that query param is pushed using the history API.

Reply:
yea i think its worth revisiting how this works, i think we might just be able to to push this down to the `FillOrder` component but keeping it the same as before to avoid refactoring too much